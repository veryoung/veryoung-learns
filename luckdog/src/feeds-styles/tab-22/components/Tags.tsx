/* eslint-disable no-loop-func */
import React from 'react';
import {
  View,
  Text,
  ScrollView,
  StyleSheet,
} from '@tencent/hippy-react-qb';
import { FeedsTheme } from './utils';
import FeedsProtect from '../../../mixins/FeedsProtect';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { CommonProps } from '../../../entity';
import { TagsBgColors } from '../../../framework/FeedsConst';

const tagsMargin = 8;
const tagsBottom = 8;

const styles = StyleSheet.create({
  imageContainer: {
    paddingBottom: 16,
    paddingHorizontal: 16,
  },
  tagsWarp: {
    flexDirection: 'column',
  },
  tagWarp: {
    flexDirection: 'row',
  },
  tag: {
    paddingLeft: 24,
    paddingRight: 24,
    paddingTop: 8,
    paddingBottom: 8,
    borderRadius: 18,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  tagText: {
    colors: FeedsTheme.SkinColor.N1,
  },
});

interface Props extends CommonProps {
  tags: string[];
  line: number;
  cardIndex: number;
}

@FeedsProtect.protect
export class Tags extends React.Component<Props> {
  public BookShelfScrollView: any;

  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'Tags');
  }

  public onScrollEndDrag = (e) => {
    const index = Math.floor(e.contentOffset.x / 160);
    this.props.doBeaconBySlide?.({ ext_data1: index });
  };

  public spliceGroup = (array, line) => {
    const length = Math.ceil(array.length / line);
    const newArray: any[] = [];
    for (let i = 1; i <= line; i++) {
      if (i !== line) {
        newArray.push(array.splice(0, length));
      } else {
        newArray.push(array.splice(0, array.length));
      }
    }
    return newArray;
  };

  public clickHandle = (tag, start) => {
    const {
      parent,
      doBeaconByClick,
    } = this.props || {};
    doBeaconByClick?.({ ext_data1: start - 1, ext_data2: tag.sTitle });
    parent.loadUrl(tag.sUrl);
  };

  public render() {
    countReRender(this, 'Tags');
    const {
      tags = [],
      line = 2,
    } = this.props || {};
    if (tags.length <= 0) {
      return null;
    }
    const tagsArray = this.spliceGroup(tags, line);
    const tagsView: any[] = [];
    const bgColorBaseIndex = Math.floor(Math.random() * 5);
    for (let i = 0; i < tagsArray.length; i++) {
      const tagView: any[] = [];
      tagsArray[i].forEach((e, j) => {
        const bgColorIndex = (bgColorBaseIndex + i + j) % 5;
        const index = i > 0 ? (i * tagsArray[i - 1].length) + j + 1 : j + 1;
        tagView.push(<View
          collapsable={false}
          style={[
            styles.tag,
            { marginRight: j === tagsArray[i].length - 1 ? 0 : tagsMargin },
            { backgroundColors: TagsBgColors[bgColorIndex] },
          ]}
          key={`${e.sTitle}-${index}`}
          onClick={() => this.clickHandle(e, index || 1)}
        >
          <Text style={styles.tagText}>{e.sTitle}</Text>
        </View>);
      });
      tagsView.push(<View
        collapsable={false}
        style={[
          styles.tagWarp,
          { marginBottom: i === tagsArray.length - 1 ? 0 : tagsBottom },
        ]}
        key={`${i}`}
      >
        {tagView}
      </View>);
    }

    return (
      <ScrollView
        contentContainerStyle={styles.imageContainer}
        horizontal
        showsHorizontalScrollIndicator={false}
        sendMomentumEvents
        ref={(ref) => {
          this.BookShelfScrollView = ref;
        }}
        onScrollEndDrag={this.onScrollEndDrag}
        scrollEventThrottle={100}
      >
        <View
          collapsable={false}
          style={styles.tagsWarp}
        >
          {tagsView}
        </View>
      </ScrollView>
    );
  }
}

