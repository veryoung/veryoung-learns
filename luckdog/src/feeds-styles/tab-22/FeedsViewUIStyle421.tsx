/**
 * @Author: veryoungwan
 * 热门分类
 */

import React from 'react';
import { View } from '@tencent/hippy-react-qb';
import { vectorToArray } from './components/utils';
import { Title } from './components';
import FeedsSpliter from '../../components/FeedsSpliter';
import FeedsViewItem from '../FeedsViewItem';
import FeedsProtect from '../../mixins/FeedsProtect';
import { reportUDS, strictExposeReporter, BusiKey } from '@/luckdog';
import { Tags } from './components/Tags';
import { CLICK_STEP, CommonCardStyle } from '../../framework/FeedsConst';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { isTopTab, throttle } from '@/luckbox';
import { scale } from '../../components/animationStyle';
import FeedsUtils from '../../framework/FeedsUtils';
import { CommonProps } from '../../entity';

interface Props extends CommonProps {
  index: number;
}

@FeedsProtect.protect
export default class FeedsViewUIStyle421 extends FeedsViewItem<Props> {
  public static getRowType() {
    return 421;
  }

  public scaleAnim = scale(200, 1.015); // 点击动画效果
  public traceid = '';
  public shouldComponentUpdate = shouldComponentUpdate(this, 'FeedsViewUIStyle421');

  public componentDidMount() {
    this.scaleAnim.onHippyAnimationEnd(() => {
      // 执行跳转
      this.clickBlankJumpMore();
    });
  }

  public UNSAFE_componentWillMount() {
    const reportInfo = FeedsUtils.getSafeProps(this.props, 'itemBean.report_info', []);
    reportInfo?.forEach(([key, value]) => {
      if (key === 'sTraceId') {
        this.traceid = value;
        return;
      }
    });
  }

  public componentWillUnmount() {
    this.scaleAnim?.destory();
  }

  /**
   * 处理跳转动画结束后加载去往更多页面
   * 跳转地址不为空
   * 支持跳转
   */
  public clickBlankJumpMore = () => {
    const { vTextLink = {} } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    const { sUrl = '' } = vectorToArray(vTextLink)[0] || {};
    if (sUrl !== '') {
      this.loadUrl(sUrl);
    }
  };

  /**
   * title点击的响应
   */
  // eslint-disable-next-line @typescript-eslint/member-ordering
  public onClickTitle = throttle(() => {
    const { clickBlankJump = false } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    if (clickBlankJump) {
      reportUDS(BusiKey.CLICK__BLANK_TO_MORE, this.props, { traceid: this.traceid });
      this.scaleAnim.start();
    }
  }, CLICK_STEP);

  public doBeaconByClick = (moreData) => {
    reportUDS(BusiKey.CLICK__CARD, this.props, moreData);
  };

  public doBeaconByClickAll = () => {
    reportUDS(BusiKey.CLICK__CARD_VIEW_ALL, this.props);
  };

  public doBeaconBySlide = (moreData = {}) => {
    reportUDS(BusiKey.SLIDE__CARD, this.props, moreData);
  };

  public onTitleLayout = (event) => {
    strictExposeReporter.updateTitleHeight(this.props.index, event.layout.height);
  };

  public onLayout = (event) => {
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.index,
      rect: event.layout,
      forceCheckExpo: true,
    });
  };

  public render() {
    countReRender(this, 'FeedsViewUIStyle421');
    const { itemBean = {}, index } = this.props;
    const { globalConf } = this.props;
    const { title = '', parsedObject = {} } = itemBean;
    const tags = JSON.parse(JSON.stringify(vectorToArray(parsedObject.vDetailData))) || [];
    const titleRight = vectorToArray(parsedObject.vTextLink)[0] || [];

    return (
      <View
        style={[{
          ...!isTopTab() && CommonCardStyle,
        }, {
          transform: [{ scale: this.scaleAnim }],
        }]}
        onLayout={this.onLayout}
      >
        <FeedsSpliter style={globalConf.style} lineStyle={itemBean.bottomLineStyle} />
        <View
          collapsable={false}
          onClick={this.onClickTitle}
        >
          <Title
            title={title}
            onLayout={this.onTitleLayout}
            parent={this}
            right={titleRight}
            doBeaconByClick={this.doBeaconByClickAll}
          />
        </View>
        <Tags
          parent={this}
          tags={tags}
          cardIndex={index}
          itemBean={itemBean}
          line={2}
          doBeaconByClick={this.doBeaconByClick}
          doBeaconBySlide={this.doBeaconBySlide}
        />
      </View>
    );
  }
}
