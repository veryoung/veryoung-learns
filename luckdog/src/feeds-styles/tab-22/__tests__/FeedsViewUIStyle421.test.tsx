import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle421 from '../FeedsViewUIStyle421';
import { commonProps } from '@test/mocks/props';
import { itemBean } from '../components/__tests__/mocks/Tags';

describe('<FeedsViewUIStyle421/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        index: 0,
        itemBean: itemBean as any,
      };
      const tree = renderer.create(<FeedsViewUIStyle421 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
