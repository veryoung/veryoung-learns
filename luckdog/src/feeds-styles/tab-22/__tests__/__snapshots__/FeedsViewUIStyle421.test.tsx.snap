// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle421/> render testing render normal 1`] = `
<div
  nativeName="View"
  onLayout={[Function]}
  style={
    Array [
      Object {},
      Object {
        "transform": Array [
          Object {
            "scale": AnimationSet {
              "animationId": undefined,
              "animationList": Array [
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
              ],
              "onAnimationEndCallback": [Function],
              "onHippyAnimationCancel": [Function],
              "onHippyAnimationEnd": [Function],
              "onHippyAnimationRepeat": [Function],
              "onHippyAnimationStart": [Function],
              "onRNfqbAnimationCancel": [Function],
              "onRNfqbAnimationEnd": [Function],
              "onRNfqbAnimationRepeat": [Function],
              "onRNfqbAnimationStart": [Function],
            },
          },
        ],
      },
    ]
  }
>
  <div
    nativeName="View"
    style={
      Array [
        Object {
          "height": 0,
        },
      ]
    }
  />
  <div
    nativeName="View"
    onClick={[Function]}
    style={
      Object {
        "collapsable": false,
      }
    }
  >
    <div
      nativeName="View"
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Array [
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "maxWidth": 252,
              "paddingBottom": 14,
              "paddingTop": 14,
            },
            Object {},
          ]
        }
        text="分类标签"
      >
        分类标签
      </p>
      <div
        className="right-view"
        nativeName="View"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "paddingBottom": 14,
            "paddingTop": 14,
            "position": "relative",
          }
        }
      >
        <p
          nativeName="Text"
          numberOfLines={1}
          style={
            Array [
              Object {
                "colors": Array [
                  "#000000",
                  "#686D74",
                  "#000000",
                  "#ffffff",
                ],
                "fontSize": 12,
                "lineHeight": 16,
              },
              Object {},
            ]
          }
          text="查看全部"
        >
          查看全部
        </p>
        <img
          alt=""
          nativeName="Image"
          noPicMode={
            Object {
              "enable": false,
            }
          }
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAbCAYAAACjkdXHAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAD6ADAAQAAAABAAAAGwAAAACSVupLAAAA40lEQVQ4EZ1UOQ4CMQyMQDQ8iIqWJ3HDAhXiEJf2KZS0PAX2B7QwRrFkuZo40sheecZJ1nZSSmkDNMACKF5vKL4Z2xJ1G+QWMMiiPmwXeORvysjRdXexe0plSGv4NsHBxCi3cgmOlMqQli7BycQoV8pmr3CmVIY0cwkuJka5U5fgSqkMaeIS3EyMcscuQS0q6TBmPUH6ANqJPfgdRmg5d3xoFWSY6DUEU4Vi6SEaOeH/zsy24b8drnO4w+bujnRvh6cqPM8rd1T6JQm/Yb4cO6b4ypEe1bajW07F8uS8AOmkovUDoSNiNZ1r/OEAAAAASUVORK5CYII="
          style={
            Object {
              "height": 9,
              "marginLeft": 4,
              "tintColors": Array [
                "#000000",
                "#686D74",
                "#000000",
                "#ffffff",
              ],
              "width": 5,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    contentContainerStyle={
      Object {
        "paddingBottom": 16,
        "paddingHorizontal": 16,
      }
    }
    horizontal={true}
    nativeName="ScrollView"
    onScrollEndDrag={[Function]}
    scrollEventThrottle={100}
    sendMomentumEvents={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      }
    }
  >
    <div
      nativeName="View"
      style={
        Array [
          Object {
            "collapse": false,
            "flexDirection": "row",
          },
          Object {
            "paddingBottom": 16,
            "paddingHorizontal": 16,
          },
        ]
      }
    >
      <div
        nativeName="View"
        style={
          Object {
            "collapsable": false,
            "flexDirection": "column",
          }
        }
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "marginBottom": 8,
              },
            ]
          }
        >
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="都市"
            >
              都市
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(239, 191, 138, 0.1)",
                    "rgba(176, 144, 108, 0.1)",
                    "#FDF8F3",
                    "#343231",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="科技修真"
            >
              科技修真
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="嘴炮"
            >
              嘴炮
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="升级流"
            >
              升级流
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(139, 206, 206, 0.1)",
                    "rgba(75, 103, 105, 0.1)",
                    "rgba(139, 206, 206, 0.1)",
                    "rgba(75, 103, 105, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="玄幻"
            >
              玄幻
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="冷静"
            >
              冷静
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(239, 191, 138, 0.1)",
                    "rgba(176, 144, 108, 0.1)",
                    "#FDF8F3",
                    "#343231",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="无限流"
            >
              无限流
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 0,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="特工"
            >
              特工
            </p>
          </div>
        </div>
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "marginBottom": 0,
              },
            ]
          }
        >
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(239, 191, 138, 0.1)",
                    "rgba(176, 144, 108, 0.1)",
                    "#FDF8F3",
                    "#343231",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="仙侠奇缘"
            >
              仙侠奇缘
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="穿越"
            >
              穿越
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="系统流"
            >
              系统流
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(139, 206, 206, 0.1)",
                    "rgba(75, 103, 105, 0.1)",
                    "rgba(139, 206, 206, 0.1)",
                    "rgba(75, 103, 105, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="热血"
            >
              热血
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                    "rgba(120, 130, 205, 0.1)",
                    "rgba(67, 73, 105, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="奇幻"
            >
              奇幻
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(239, 191, 138, 0.1)",
                    "rgba(176, 144, 108, 0.1)",
                    "#FDF8F3",
                    "#343231",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="重生"
            >
              重生
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 8,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                    "rgba(255, 121, 110, 0.1)",
                    "rgba(121, 69, 67, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="轻松"
            >
              轻松
            </p>
          </div>
          <div
            nativeName="View"
            onClick={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 18,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 8,
                },
                Object {
                  "marginRight": 0,
                },
                Object {
                  "backgroundColors": Array [
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                    "rgba(76, 178, 255, 0.1)",
                    "rgba(49, 92, 125, 0.1)",
                  ],
                },
              ]
            }
          >
            <p
              nativeName="Text"
              style={
                Object {
                  "colors": Array [
                    "#000000",
                    "#686D74",
                    "#000000",
                    "#ffffff",
                  ],
                }
              }
              text="赚钱"
            >
              赚钱
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
